server/src/main/java/org/spine3/server/reflect/Field.java

```diff
@@ -0,0 +1,186 @@
+        checkNotNull(filter);
+        final String fieldName = getFieldName(filter);
+        return newField(messageClass, fieldName);
+    }
+
+    private static String getFieldName(FieldFilter filter) {
+        final String fieldPath = filter.getFieldPath();
+        final String fieldName = fieldPath.substring(fieldPath.lastIndexOf('.') + 1);
+
+        if (fieldName.isEmpty()) {
+            throw newIllegalArgumentException(
+                    "Unable to get a field name from the field filter: %s",
+                    filter);
+        }
+        return fieldName;
+    }
+
+    /**
+     * Returns the class of the Protobuf message field.
+     *
+     * @param field the field descriptor
+     * @return the class of the field
+     * @throws IllegalArgumentException if the field type is unknown
+     */
+    @SuppressWarnings("OverlyComplexMethod")    // as each branch is a fairly simple.
+    public static Class<?> getFieldClass(FieldDescriptor field) {
+        checkNotNull(field);
+        final FieldDescriptor.JavaType javaType = field.getJavaType();
+        final TypeUrl typeUrl;
+        switch (javaType) {
+            case INT:
+                return Integer.class;
+            case LONG:
+                return Long.class;
+            case FLOAT:
+                return Float.class;
+            case DOUBLE:
+                return Double.class;
+            case BOOLEAN:
+                return Boolean.class;
+            case STRING:
+                return String.class;
+            case BYTE_STRING:
+                return ByteString.class;
+            case ENUM:
+                typeUrl = TypeUrl.from(field.getEnumType());
+                final Class<? extends Message> enumClass = typeUrl.getJavaClass();
+                return enumClass;
+            case MESSAGE:
+                typeUrl = TypeUrl.from(field.getMessageType());
+                final Class<? extends Message> msgClass = typeUrl.getJavaClass();
+                return msgClass;
+            default:
+                throw newIllegalArgumentException("Unknown field type discovered: %s",
+                                                   field.getFullName());
+        }
+    }
+
+    /**
+     * Obtains the name of the field.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Obtains the value of the field in the passed object.
+     *
+     * <p>If the corresponding field is of type {@code Any} it will be unpacked.
+     *
+     * @throws IllegalStateException if getting the field value caused an exception.
+     *                               The root cause will be available from the thrown instance.
+     */
+    public Message getValue(Message object) {
+        final Message fieldValue;
+        final Message result;
+        try {
+            fieldValue = (Message) getter.invoke(object);
+
+            result = fieldValue instanceof Any
+                     ? AnyPacker.unpack((Any) fieldValue)
```

Comment:
Please check what happens if the value is `Any.getDefaultInstance()`. If I remember it right, the `AnyPacker` will throw an exception in that case.